test/test_profiler.py

```diff
@@ -138,6 +140,182 @@ def test_json_output():
     assert len(root_frame["children"]) == 2
 
 
+def test_speedscope_output():
+    with fake_time():
+        with Profiler() as profiler:
+            long_function_a()
+            long_function_b()
+
+    output_data = profiler.output(renderers.SpeedscopeRenderer())
+
+    output = json.loads(output_data)
+
+    file_level_schema_fields = {
+        "$schema",
+        "name",
+        "exporter",
+        "activeProfileIndex",
+        "profiles",
+        "shared",
+    }
+    for file_field in file_level_schema_fields:
+        assert file_field in output
+
+    assert output["$schema"] == "https://www.speedscope.app/file-format-schema.json"
+    assert "pyinstrument" in output["exporter"]
+    assert output["activeProfileIndex"] is None
+    assert "CPU profile" in output["name"]
+
+    assert "frames" in output["shared"]
+    speedscope_frame_list = output["shared"]["frames"]
+
+    # Distinct functions called stores indices in key-value pairs because function
+    # index lookup needed for Speedscope event list tests. Were we not testing
+    # the event list, the distinct functions called could be stored as a tuple.
+    distinct_functions_called = {
+        "test_speedscope_output": 0,
+        "long_function_a": 1,
+        "sleep": 2,
+        "long_function_b": 3,
+    }
+    assert len(speedscope_frame_list) == len(distinct_functions_called)
+    speedscope_frame_fields = tuple([field.name for field in dataclasses.fields(SpeedscopeFrame)])
+    for (function_name, frame_index) in distinct_functions_called.items():
+        for frame_field in speedscope_frame_fields:
+            assert frame_field in speedscope_frame_list[frame_index], (
+                f"Field named '{frame_field}' not in Speedscope output"
+                f"at 'shared.frames[{frame_index}]'"
```

Comment:
I really appreciate the craft and effort in these assert messages! But when you're running a test with pytest, and an assert fails, pytest will output a lot of useful information that is relevant to that fail, including the contents of variables etc. So it's probably not worth including that information as an assert message, I think. Also, it will help the test be easier to read.

Reply:
Removing them will also reduce the need for a lot of local variables, which should help readability a bit.

Reply:
Works for me.

Reply:
Removed assertion messages and associated local variables in [`97925fb2`](https://github.com/joerick/pyinstrument/pull/160/commits/97925fb284a34a111b7b99bbef8cd0bb72e7c584).